### PR TITLE
fix(docs): improve v5 migration guide

### PR DESCRIPTION
**Pull request by dai-shi**
https://github.com/pmndrs/zustand/issues/2741#issuecomment-2359569305

### CODE REVIEW

File: docs/migrations/migrating-to-v5.md [hunk 1]
```diff
@@ -109,6 +110,26 @@ The error message will be something like this:
 Uncaught Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
 ```
 
+To fix it, `useShallow` should help return a stable reference.
```
Comment (sewera) [inline docs/migrations/migrating-to-v5.md:113]:
```suggestion
To fix it, use the `useShallow` hook, which will return a stable reference.
```

File: docs/migrations/migrating-to-v5.md [hunk 2]
```diff
@@ -109,6 +110,26 @@ The error message will be something like this:
...
+```js
+// v5
+import { useShallow } from 'zustand/shallow'
+
+const [searchValue, setSearchValue] = useStore(
+  useShallow((state) => [state.searchValue, state.setSearchValue]),
+)
+```
+
+Here's another example that may cause infinite loops.
```
Comment (sewera) [inline docs/migrations/migrating-to-v5.md:124]: I don't really understand why there is "another example," but the comment says that it's v4. Didn't you mean to say "Here's a similar example in v4 that may cause infinite loops?"

**Review by sewera — CHANGES_REQUESTED** [review]

File: docs/migrations/migrating-to-v5.md (hunk 2 above)
Comment (dai-shi) [reply docs/migrations/migrating-to-v5.md:124]: I mean there are several cases that cause infinite loops.
I want to show typical two examples that are actually reported by users.

**Review by sewera — APPROVED** [review]

File: docs/migrations/migrating-to-v5.md (hunk 2 above)
Comment (dai-shi) [reply docs/migrations/migrating-to-v5.md:124]: Feel free to make suggestions to clarify that.

**Pull request merged**